modules/uid2IdSystem_shared.js

```diff
@@ -1,17 +1,109 @@
 /* eslint-disable no-console */
+import { ajax } from '../src/ajax.js';
+
 export const Uid2CodeVersion = '1.1';
 
 function isValidIdentity(identity) {
   return !!(typeof identity === 'object' && identity !== null && identity.advertising_token && identity.identity_expires && identity.refresh_from && identity.refresh_token && identity.refresh_expires);
 }
 
+function isNormalizedPhone(phone) {
+  return /^\+[0-9]{10,15}$/.test(phone);
+}
+
+function isBase64Hash(value) {
+  if (!(value && value.length === 44)) {
+    return false;
+  }
+
+  try {
+    return btoa(atob(value)) === value;
+  } catch (err) {
+    return false;
+  }
+}
+
+function isCSTGOptions(
+  maybeOpts,
+  _logWarn
+) {
+  if (typeof maybeOpts !== 'object' || maybeOpts === null) {
+    _logWarn('CSTG opts must be an object');
+    return false;
+  }
+
+  const opts = maybeOpts;
+  if (typeof opts.serverPublicKey !== 'string') {
+    _logWarn('CSTG opts.serverPublicKey must be a string');
+    return false;
+  }
+  const serverPublicKeyPrefix = /^UID2-X-[A-Z]-.+/;
+  if (!serverPublicKeyPrefix.test(opts.serverPublicKey)) {
+    _logWarn(
+      `CSTG opts.serverPublicKey must match the regular expression ${serverPublicKeyPrefix}`
+    );
+    return false;
+  }
+  // We don't do any further validation of the public key, as we will find out
+  // later if it's valid by using importKey.
+
+  if (typeof opts.subscriptionId !== 'string') {
+    _logWarn('CSTG opts.subscriptionId must be a string');
+    return false;
+  }
+  if (opts.subscriptionId.length === 0) {
+    _logWarn('CSTG opts.subscriptionId is empty');
+    return false;
+  }
+  return true
+}
+
+function getValidIdentity(opts, _logWarn) {
+  if (opts.emailHash) {
+    if (!isBase64Hash(opts.emailHash)) {
+      _logWarn('CSTG opts.emailHash is invalid');
+      return;
+    }
+    return { email_hash: opts.emailHash }
+  }
+
+  if (opts.phoneHash) {
+    if (!isBase64Hash(opts.phoneHash)) {
+      _logWarn('CSTG opts.phoneHash is invalid');
+      return;
+    }
+    return { phone_hash: opts.phoneHash }
+  }
+
+  if (opts.email) {
+    const normalizedEmail = normalizeEmail(opts.email);
+    if (normalizedEmail === undefined) {
+      _logWarn('CSTG opts.email is invalid');
+      return;
+    }
+    return { email: opts.email }
+  }
+
+  if (opts.phone) {
+    if (!isNormalizedPhone(opts.phone)) {
+      _logWarn('CSTG opts.phone is invalid');
+      return;
+    }
+    return { phone: opts.phone }
+  }
+}
+
 // This is extracted from an in-progress API client. Once it's available via NPM, this class should be replaced with the NPM package.
 export class Uid2ApiClient {
   constructor(opts, clientId, logInfo, logWarn) {
     this._baseUrl = opts.baseUrl;
     this._clientVersion = clientId;
     this._logInfo = logInfo;
     this._logWarn = logWarn;
+    if (opts.cstg) {
```

Comment:
Just noting a difference with the JS SDK. In the SDK we pass the CSTG config to the method, not to the constructor.

Reply:
When using JS SDK, publisher could call `setIdentityFromEmail` and `setIdentityFromPhone` to get identity. When they use Prebid.js they should only call `setConfig` when they want to use a userId https://docs.prebid.org/dev-docs/modules/userId.html#how-it-works